libs/state/src/lib/rx-state.service.ts

```diff
@@ -394,32 +396,51 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
     signal: Signal<V>,
     projectSliceFn: ProjectValueReducer<T, K, V>
   ): void;
+
+  connect<K extends keyof T>(
+    object: Partial<{ [K: string]: Observable<T[K]> | Signal<T[K]> }>
+  ): void;
+
   /**
    * @internal
    */
   connect<K extends keyof T, V extends Partial<T>>(
-    keyOrInputOrSlice$: K | Observable<Partial<T> | V> | Signal<Partial<T> | V>,
+    keyOrInputOrSliceOrPartials$:
+      | K
+      | Observable<Partial<T> | V>
+      | Signal<Partial<T> | V>
+      | Partial<{ [K: string]: Observable<T[K]> | Signal<T[K]> }>,
     projectOrSlices$?:
       | ProjectStateReducer<T, V>
       | Observable<T[K] | V>
       | Signal<T[K] | V>,
     projectValueFn?: ProjectValueReducer<T, K, V>
   ): void {
+    if (
+      isPartialOfSignalsOrObservablesGuard(keyOrInputOrSliceOrPartials$) &&
+      !projectOrSlices$ &&
+      !projectValueFn
+    ) {
+      this.accumulator.nextSliceObservable(
+        convertPartialsToObservable(keyOrInputOrSliceOrPartials$, this.injector)
+      );
+      return;
+    }
     let inputOrSlice$: Observable<Partial<T> | V> | undefined;
-    if (!isKeyOf<T>(keyOrInputOrSlice$)) {
-      if (isObservable(keyOrInputOrSlice$)) {
-        inputOrSlice$ = keyOrInputOrSlice$;
+    if (!isKeyOf<T>(keyOrInputOrSliceOrPartials$)) {
+      if (isObservable(keyOrInputOrSliceOrPartials$)) {
+        inputOrSlice$ = keyOrInputOrSliceOrPartials$;
       } else {
         // why can't typescript infer the correct type?
         inputOrSlice$ = toObservable(
-          keyOrInputOrSlice$ as Signal<Partial<T> | V>,
+          keyOrInputOrSliceOrPartials$ as Signal<Partial<T> | V>,
           { injector: this.injector }
         );
       }
     }
```

Comment:
I think this nested if/else branching starts to be hard to read. I think there's a better way to handle overloads, would be nice to refactor it at some point.

Reply:
Totally agree. Overloads in Typescript are a complete mess

Reply:
@edbzn I started in a separate PR to simplify the `connect`-method: https://github.com/rx-angular/rx-angular/pull/1652 
wdyt? 
To me it looks better. However it is not finished yet. There are two failing test cases bc my guards are not perfect yet I believe